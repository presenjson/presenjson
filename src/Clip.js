import React from 'react';
import PropTypes from 'prop-types';

const Clip = (props) => {
    return (<div className='clip'>
        {props.children}
    </div>);
}

Clip.propTypes = {
    length: PropTypes.number
};

Clip.defaultProps = {
    length: 0
}

export default Clip;
