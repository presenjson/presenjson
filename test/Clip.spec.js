import React from 'react';
import { shallow } from 'enzyme';
import Clip from '../src/Clip';

describe('Clip', () => {
    it('renders nothing', () => {
        expect(
            shallow(
                <Clip>
                    <div className="test" />
                </Clip>
            )
        ).toMatchSnapshot();
    });
    it('sets default props', () => {
        expect(Clip.defaultProps).toEqual({ length: 0, delay: 0 });
    });
});
