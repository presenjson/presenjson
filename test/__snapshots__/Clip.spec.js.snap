// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clip renders nothing 1`] = `""`;
